From f77bbd8a0e51bed3c7eaf2397e80328d929e84be Mon Sep 17 00:00:00 2001
From: Jaroslav Kysela <jkysela@redhat.com>
Date: Wed, 3 Sep 2014 07:20:07 -0400
Subject: [alsa] usb-audio: work around KEF X300A firmware bug

Message-id: <140972823573-125-patchset-jkysela@redhat.com>
Patchwork-id: 90825
O-Subject: [RHEL 7.1 PATCH 124/442] ALSA: usb-audio: work around KEF X300A firmware bug
Bugzilla: 1112200
RH-Acked-by: John Linville <linville@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Ivan Vecera <ivecera@redhat.com>

When the driver tries to access Function Unit 10, the KEF X300A
speakers' firmware apparently locks up, making even PCM streaming
impossible.  Work around this by ignoring this FU.

Cc: <stable@vger.kernel.org>
Signed-off-by: Clemens Ladisch <clemens@ladisch.de>
Signed-off-by: Takashi Iwai <tiwai@suse.de>

(picked from kernel: 624aef494f86ed0c58056361c06347ad62b26806)
rhbz#1112200
Signed-off-by: Jaroslav Kysela <jkysela@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/sound/usb/mixer_maps.c b/sound/usb/mixer_maps.c
index 32af6b7..d1d72ff 100644
--- a/sound/usb/mixer_maps.c
+++ b/sound/usb/mixer_maps.c
@@ -328,6 +328,11 @@ static struct usbmix_name_map gamecom780_map[] = {
  {}
 };
 
+static const struct usbmix_name_map kef_x300a_map[] = {
+ { 10, NULL }, /* firmware locks up (?) when we try to access this FU */
+ { 0 }
+};
+
 /*
  * Control map entries
  */
@@ -419,6 +424,10 @@ static struct usbmix_ctl_map usbmix_ctl_maps[] = {
   .id = USB_ID(0x200c, 0x1018),
   .map = ebox44_map,
  },
+ {
+  .id = USB_ID(0x27ac, 0x1000),
+  .map = kef_x300a_map,
+ },
  { 0 } /* terminator */
 };
 
-- 
1.7.1